homeassistant/components/zwave_me/climate.py

```diff
@@ -0,0 +1,108 @@
+"""Representation of a thermostat."""
+from __future__ import annotations
+
+from zwave_me_ws import ZWaveMeData
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ZWaveMeEntity
+from .const import DOMAIN, ZWaveMePlatform
+
+TEMPERATURE_DEFAULT_STEP = 0.5
+
+DEVICE_NAME = ZWaveMePlatform.CLIMATE
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the climate platform."""
+
+    @callback
+    def add_new_device(new_device: ZWaveMeData) -> None:
+        """Add a new device."""
+        controller = hass.data[DOMAIN][config_entry.entry_id]
+        climate = ZWaveMeClimate(controller, new_device)
+
+        async_add_entities(
+            [
+                climate,
+            ]
+        )
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(
+            hass, f"ZWAVE_ME_NEW_{DEVICE_NAME.upper()}", add_new_device
+        )
+    )
+
+
+class ZWaveMeClimate(ZWaveMeEntity, ClimateEntity):
+    """Representation of a ZWaveMe sensor."""
+
+    def set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        if (temperature := kwargs.get(ATTR_TEMPERATURE)) is None:
+            return
+
+        self.hass.data[DOMAIN].zwave_api.send_command(
+            self.device.id, "exact?level=" + str(temperature)
+        )
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the temperature_unit."""
+        return self.device.scaleTitle
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the state of the sensor."""
+        return self.device.level
+
+    @property
+    def max_temp(self) -> float:
+        """Return min temperature for the device."""
+        return self.device.max
+
+    @property
+    def min_temp(self) -> float:
+        """Return max temperature for the device."""
+        return self.device.min
+
+    @property
+    def hvac_modes(self) -> list[str]:
```

Comment:
How do we change between these modes? There's no service handler method `set_hvac_mode` implemented?

Reply:
This is not yet well supported in Z-Way. We are reworking this part and once done it will be reflected in this integration too. Currently modes are only changeable in the Expert UI of Z-Way

Reply:
If we can't change the modes, we should probably just return a list with the heat mode here.

Please test how this behaves in the frontend.